---
title: "GO Enrichment Analysis"
output: html_document
author: "Rashika Ranasinghe"
date: "2025-11-24"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Perform a GO Enrichment Analysis for the genes in the biggest inversion in the human genome, chr8p23.1, approximately 4.5 to 12 megabases (Mb) in size.

```{r cars}
# Load the required libraries
library(biomaRt)
library(gggenes)
library(ggplot2)
library(dplyr)
library(org.Hs.eg.db)
library(clusterProfiler)
library(enrichplot)
library(Gviz)
```

Pull genes that are on the inversion region from Ensembl
```{r}
# Define inversion coordinates
chromosome <- "8"
inv_start <- 7370695
inv_end   <- 11678224

# Connect to Ensembl
mart <- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl")

# Query Ensembl for genes in the inversion
genes <- getBM(
  attributes = c("ensembl_gene_id", "external_gene_name", "description",
    "chromosome_name", "start_position", "end_position", "gene_biotype"),
  filters = c("chromosome_name", "start", "end"),
  values = list(chromosome, inv_start, inv_end),
  mart = mart
)
```


Make a diagram of all the genes on the inversion region
```{r  fig.width=10, fig.height=10}
# Prepare the data
gene_plot_df <- genes
gene_plot_df$gene_name <- gene_plot_df$external_gene_name
gene_plot_df$start <- gene_plot_df$start_position
gene_plot_df$end   <- gene_plot_df$end_position

# Plot genes along the chromosome
ggplot(gene_plot_df, aes(
  xmin = start,
  xmax = end,
  y = gene_biotype,     # improvement: separate rows by biotype
  fill = gene_biotype,
  label = gene_name
)) +
  geom_gene_arrow(arrowhead_width = unit(2, "mm")) +
  geom_text(aes(x = (start + end)/2, y = as.numeric(as.factor(gene_biotype)) + 0.2),
            size = 2, angle = 90, hjust = 0) +
  scale_fill_brewer(palette = "Set3") +
  ggtitle("Genes in 8p23.1 Inversion - Human Genome") +
  theme_bw() +   labs(x = "Chromosomal Position")   +
  theme(legend.position = "bottom")

```

GO enrichment analysis (For all biotypes)
```{r}
# Convert to ENTREZ
entrez_ids <- bitr(
  genes$external_gene_name,
  fromType = "SYMBOL",
  toType = "ENTREZID",
  OrgDb = org.Hs.eg.db
)$ENTREZID

enrichFrame <- enrichGO(
  gene = entrez_ids,
  OrgDb = org.Hs.eg.db,
  keyType = "ENTREZID",
  ont = "ALL",              
  pvalueCutoff = 0.05,
  readable = TRUE
)

# Save enrichment table
#write.csv(as.data.frame(ego), "GO_enrichment_chr8p23.1.csv", row.names = FALSE)
```

Plot the analysis results
```{r fig.width=10, fig.height=7}
# 1. Bar plot
barplot(enrichFrame,
        x = "GeneRatio", # plot the gene ratio = # gene of that enrichment / toal # genes
        color = "p.adjust",
        title = "GO Enrichment (n = 31)",
        showCategory = 31, # plot all the GO enrichments
        label_format = 80) + theme(
plot.margin = margin(10, 1, 1, 10) # Top, Right, Bottom, Left margins in points
)

```


```{r fig.width=10, fig.height=7}
# 2. Dot plot
dotplot(enrichFrame,
        x = "Count",
        color = "p.adjust",
        title = "GO Enrichment (n = 31)",
        showCategory = 31,
        label_format = 80) +
  theme(plot.margin = margin(20, 20, 20, 20))

```
split enrichment results by ontology

```{r fig.width=8, fig.height=10}
# make the enrichFrame object a dataframe 
enrich_df <- as.data.frame(enrichFrame)

# make seperate vectors for each ontology group
ego_BP <- enrich_df %>% filter(ONTOLOGY == "BP")
ego_CC <- enrich_df %>% filter(ONTOLOGY == "CC")
ego_MF <- enrich_df %>% filter(ONTOLOGY == "MF")
table(enrich_df$ONTOLOGY)

# Plot the data
dotplot(enrichFrame, split = "ONTOLOGY") +
  facet_grid(ONTOLOGY ~ ., scales = "free",
             labeller = labeller(ONTOLOGY =  c("BP" = "Biological Process",
                                               "CC" = "Cellular Component",
                                               "MF" = "Molecular Function"))) +
  ggtitle("GO Enrichment Separated by Ontology (BP, CC, MF)")
```


```{r fig.width=10, fig.height=7}
# 3. Category Netplot
# Helpful to see which genes are involved in enriched pathways and genes that may belong to multiple annotation categories
cnetplot(enrichFrame,
         categorySize = "pvalue",
         showCategory = 31)

```


